capsules/src/virtual_spi.rs

```diff
@@ -106,9 +138,18 @@ impl<'a, SPI: hil::spi::SpiMaster> SPIMasterDevice<'a, SPI> {
         self.mux.devices.push_head(self);
         self.client.set(Some(client));
     }
+
+    // Binding virtualization and configuration causes problems,
+    // because it implies that a virtualized client never wants to
+    // use more than one chip select line. Counter-example is the
```

Comment:
This implies that there are N SPI system call interfaces, where N is the number of chip selects. The downside is that you have to allocate memory for all of them. Wouldn't it make more sense to just limit the chip selects that can be used by the chip_selects parameter? E.g., line 255?

I'd argue that encapsulating the chip select in the virtualization was seen as useful and needed for one application (a particular driver), but doesn't generalize to user-land control of the SPI.